rfd/0134-cloud-first-deployment.md

```diff
@@ -0,0 +1,213 @@
+
+With this in mind, the Cloud version will consist of:
+
+- Major: "development" version number on master branch (latest stable self-hosted release + 1).
+- Minor/patch: always 0.
+- Prerelease: `-cloud.X.Y` where `X` will represent a timestamp of when the tag
+  was cut and `Y` the corresponding commit hash: `v14.0.0-cloud.20230712T123633.aabbcc`.
+
+Once it is time for the next self-hosted major release (every 3-4 months), the
+new release branch will cut and the Cloud release version major component will
+increment.
+
+```
+                                                   14.0.0     14.0.1     14.0.2
+                                        branch/v14 |----------|----------|---------....---->
+                                                   |
+                                                   |
+            14.0.0-cloud.20230712T123633.aabbcc      14.0.0-cloud...  15.0.0-cloud...    15.0.0-cloud...    16.0.0-cloud...
+master -----|-------....---------------------------|------------------|-------....-------|------------------|--------------....---->
+                                                                      |
+                                                                      |
+                                                           branch/v15 |---------|---------|---------...---->
+                                                                      15.0.0    15.0.1    15.0.2
+```
+
+Once a self-hosted release branch is cut, subsequent self-hosted minor/patch
+releases are published without any changes to the current process, but on a
+less frequent cadence than Cloud releases off of master. Changes to the release
+branches will follow the same "develop on master, backport to release branch"
+workflow we're using now.
+
+### Cloud release artifacts
+
+The self-hosted release and promotion process is slow and takes 2-3 hours
+because it builds a full suite of artifacts. Cloud users need most of these
+artifacts as well for the agents they run on their own infrastructure, however
+Cloud control plane needs a very limited subset of all artifacts. For this
+reason, the Cloud release and promotion process will be split in 2 parts:
+
+- The workflow that builds artifacts required for the control plane: amd64
+  binaries and Docker images.
+- The workflow that builds remaining artifacts such as deb/rpm packages and
+  Helm charts.
+
+This will allow us to produce rapid releases that affect only control plane
+services (auth, proxy) while still having the ability to push remaining artifacts
+when needed.
+
+Both workflows will run in Github Actions and will reuse existing bits of the
+full self-hosted release pipeline.
+
+### Cloud promotion process
+
+The Cloud release promotion process will be staged, with a different class of
+tenants being upgraded at each stage:
+
+1. Canary
+2. Internal
+3. Trials
+4. Team & Enterprise
+
+The canary tenants will be a set of precreated clusters with some data in them
```

Comment:
Canaries are typically a subset of real users with real workloads, not a testing environment. As described here, this does not sound like a canary candidate. Should this be renamed to "Testing" or something similar?